core/src/subgraph/runner.rs

```diff
@@ -880,197 +1077,247 @@ where
     C: Blockchain,
     T: RuntimeHostBuilder<C>,
 {
-    async fn handle_process_block(
+    async fn handle_process_wasm_block(
```

Comment:
There seems to be _a lot_ of code duplication between `handle_process_wasm_block` and `handle_process_block` which I fear will lead to subtle bugs as that logic evolves over time. Is there no way to avoid that duplication? I think `handle_process_block` would also be improved by introducing some helper structs that encapsulate some of the logic that is shared between the two; I know it's grown over time, but to the uninitiated, the code for `handle_process_block` is very hard to follow.

Reply:
Can you then add an issue so this doesn't get completely forgotten?